src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets

```diff
@@ -23,10 +23,120 @@ Copyright (c) .NET Foundation. All rights reserved.
+   </Target>
+  <!--
+    ============================================================
+                                        CacheWorkerMapper
+
+   Spins of the compose mappers
+    ============================================================
+    -->
+
+  <Target Name="CacheWorkerMapper">
+    
+    <MSBuild Projects="$(MSBuildProjectFullPath)"
+                 Targets="CacheWorker"
+                 Properties="CachePackageName=%(PackageReference.Identity);
+                             CachePackageVersion=%(PackageReference.Version);
+                             ComposeWorkingDir=$(ComposeWorkingDir);
+                             PublishDir=$(PublishDir);
+                             TargetFramework=$(TargetFramework);
+                             JitPath=$(JitPath);
+                             Crossgen=$(Crossgen);
+                             SkipUnchangedFiles=$(SkipUnchangedFiles);
+                             PreserveCacheLayout=$(PreserveCacheLayout)"
+                  ContinueOnError="WarnAndContinue"
+             />
+  </Target>
+
+  <Target Name="CacheWorker"
+          DependsOnTargets="_InitializeBasicProps;
+                            PrepforRestoreForComposeCache;
+                            CacheWorkerPerformWork"/>
   
+  <Target Name="CacheWorkerPerformWork"
+          DependsOnTargets="RestoreForComposeCache;
+                            ComputeAndCopyFilesToCacheDirectory;"
+          Condition="!Exists($(CacheWorkerWorkingDir))" />
+
+<!--
+    ============================================================
+                                        CacheFinalizer
+
+   Cleansup and produces artifacts after Completion of cache
+    ============================================================
+    -->
+  
+  <Target Name="CacheFinalizer">
+    
+    <ItemGroup>
+      <ListOfPackageReference Include="@(AllResolvedPackagesPublished -> '%20%20%20%20%20&lt;PackageReference Include=&quot;%(Identity)&quot;  Version =&quot;%(Version)&quot;/&gt;')"/>
+    </ItemGroup>
+    <PropertyGroup>
+     <_CacheArtifactContent>
+      <![CDATA[
+<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="$(MSBuildToolsVersion)">
+  <ItemGroup>
+@(ListOfPackageReference)
+  </ItemGroup>
+</Project>    
+]]>
+       </_CacheArtifactContent>
+      </PropertyGroup>
+    <WriteLinesToFile
+             File="$(CacheArtifactXml)"
+             Lines="$(_CacheArtifactContent)"
+             Overwrite="true" />
+
+    <Message Text="Files were composed in $(PublishDir)"
+                 Importance="high"/>
+    <Message Text="The list of packages cached is in $(CacheArtifactXml) "
+                 Importance="high"/>
+    <RemoveDir
+        Condition="'$(PreserveComposeWorkingDir)' != 'true'"
+        Directories="$(ComposeWorkingDir)"
+        ContinueOnError="WarnAndContinue"/>
+  </Target>
+
+  <Target Name="_InitializeBasicProps">
+    <PropertyGroup>
+      <PathSeparator>$([System.IO.Path]::PathSeparator)</PathSeparator>
```

Comment:
This property is not used in this file.  It is wrong to declare it here, if this whole file doesn't use it.

Same for `DirectorySeparatorChar`.